src/libraries/System.Private.CoreLib/src/System/IntPtr.cs

```diff
@@ -17,8 +18,9 @@
 namespace System
 {
     [Serializable]
+    [StructLayout(LayoutKind.Sequential)]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    public readonly struct IntPtr : IEquatable<IntPtr>, ISerializable
+    public readonly struct IntPtr : IEquatable<IntPtr>, IComparable, IComparable<IntPtr>, IFormattable, ISerializable
```

Comment:
There are number of places throughout CoreLib that assume IntPtr/UIntPtr are not comparable. We should fix them up as necessary.

For example, https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Private.CoreLib/src/System/Array.cs#L513 should be fixed up.

Reply:
@jkotas, do we have a rough list of these places?

It looks like 2 places in System.Array and no others for `ELEMENT_TYPE_I` or `ELEMENT_TYPE_U` (in managed code, I didn't check native):
* https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Array.cs#L515
* https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Array.cs#L1694

I did a quick scan of some other places and didn't spot anything that was keying off `typeof(IntPtr` or `typeof(UIntPtr)`. Most places look to just key off `x is IComparable`.